service/history/historyEngine_test.go

```diff
@@ -3740,10 +3740,10 @@ func (s *engineSuite) TestRequestCancel_RespondDecisionTaskCompleted_Scheduled()
 	s.Nil(err)
 
 	executionBuilder := s.getBuilder(domainID, we)
-	s.Equal(int64(11), executionBuilder.GetExecutionInfo().NextEventID)
+	s.Equal(int64(12), executionBuilder.GetExecutionInfo().NextEventID)
 	s.Equal(int64(7), executionBuilder.GetExecutionInfo().LastProcessedEvent)
 	s.Equal(persistence.WorkflowStateRunning, executionBuilder.GetExecutionInfo().State)
-	s.False(executionBuilder.HasPendingDecisionTask())
+	s.True(executionBuilder.HasPendingDecisionTask())
```

Comment:
Can we also assert on expected history after processing of decision?
Also can you add a test case when completion and request_cancel happens in the same transaction.  We want to make sure workflow completion is the last event in this situation.